scripts/utilities/rescal_utilities.py

```diff
@@ -5,381 +5,474 @@
+
+    def get(self, option):
+        return self.options[option]
+
+    def set_header(self, header):
+        self.header_body = header
+
+    def set(self, option_dict):
+        # Takes a dictionary of {"option name" : value} pairs
+        for option in option_dict.keys():
+            if option in self.options.keys():
+                self.options[option] = option_dict[option]
+            else:
+                print("Skipping nonexistent option " + option)
+
+    def __write_run_rescal(self, f):
+        # Sub-function of write() that writes the call to rescal
+        # with appropriate flags
+
+        f.write('# ----Rescal----\n')
+        if self.options['nice']:
+            f.write('nice ')
+        f.write("./rescal $PAR_FILE")
+        for option in self.__flag_options.keys():
+            if (self.options[option] == True):
+                # -flag
+                f.write(" -" + self.__flag_options[option])
+            elif self.options[option]: # == any value except False or True
+                # -flag VALUE
+                f.write(" -" + str(self.__flag_options[option]) + " " + str(self.options[option]))
+        f.write("\n \n")
+
+
+    def write(self, filename):
+        with open(filename, "w") as f:
+            f.write("#!/bin/bash \n \n")
+            f.write("################## \n ## ReSCAL run script ## \n################")
+            f.write("\n \n ##" + self.header_start + "\n ##" + self.header_body + "\n \n")
+
+            # Organizational options
+            f.write("# ----Organizational tasks---- \n")
+            if self.options['clean']:
+                f.write("# Remove files from previous runs \n")
+                f.write("./clean \n\n")
+            if self.options['backup']:
+                f.write("# Make an archive from the sources \n")
+                f.write("./dobackup \n\n")
+
+
+
+            #Linking - not optional
+            f.write("if [ ! -e genesis ]; then \n  ln -s " + self.options['rescallocation'] +  "/genesis . \nfi \n")
+            f.write("if [ ! -e rescal ]; then \n  ln -s "  + self.options['rescallocation'] + "/rescal . \nfi \n\n")
+            f.write("ln -s " + self.options['rescallocation'] + "/rescal-ui.xml . \n \n")
+
+            #Parameter file
+            f.write("# ----Parameter file----\n")
+            f.write("PAR_FILE=\"" + self.options['parfile'] + "\"\n\n")
+            f.write("echo PAR_FILE=$PAR_FILE\n\n")
+
+            f.write('# -----Physical properties----\n')
+            f.write("cp -r " + self.options['real_data_location'] + " . \n\n")
+
+            # Run options
+            f.write("# ----Run options----\n")
+            f.write("export OMP_NUM_THREADS=1 \n")
+            f.write("GENESIS_LOG_FILE=\"" + self.options['genesislog'] + "\"\n")
+            f.write("RESCAL_LOG_FILE=\""  + self.options['rescallog']  + "\"\n\n")
+
+            # Run genesis if no premade .csp file
+            if  not self.options['premade_csp']:
```

Comment:
this is where the option to use a premade .csp files get used instead of calling genesis